Sources/SwiftSyntaxBuilder/BuildableNodes.swift.gyb

```diff
@@ -95,28 +100,31 @@ public struct ${type.buildable()}${conformance_clause(conformances)} {
   ///  - Initializing syntax collections using result builders
   ///  - Initializing tokens without default text using strings
   public init(
+    leadingTrivia: Trivia = [],
     ${',\n    '.join(convenience_init_normal_parameters + convenience_init_result_builder_parameters)}
   ) {
     self.init(
+      leadingTrivia: leadingTrivia,
       ${',\n      '.join(delegated_init_args)}
     )
   }
 %   end
 
-  func build${type.base_name()}(format: Format, leadingTrivia: Trivia? = nil) -> ${type.syntax()} {
+  func build${type.base_name()}(format: Format, leadingTrivia additionalLeadingTrivia: Trivia? = nil) -> ${type.syntax()} {
     let result = SyntaxFactory.make${type.base_name()}(
       ${',\n      '.join(['%s: %s' % (child.name(), child.generate_expr_build_syntax_node(child.name(), 'format')) for child in children])}
     )
-    if let leadingTrivia = leadingTrivia {
-      return result.withLeadingTrivia(leadingTrivia + (result.leadingTrivia ?? []))
-    } else {
+    let combinedLeadingTrivia = leadingTrivia + (additionalLeadingTrivia ?? []) + (result.leadingTrivia ?? [])
+    if combinedLeadingTrivia.isEmpty {
```

Comment:
Is this `if`-statement really necessary or can we just 
```swift
return result.withLeadingTrivia(combinedLeadingTrivia)
```

Reply:
Yeah, we could probably do that, I wasn't sure whether there was some internal overhead to attaching potentially empty trivia to every node.

Reply:
I don’t think we should need to worry about micro-performance-optimization in SwiftSyntaxBuilder.